lib/android.js

```diff
@@ -1378,16 +1378,53 @@ function makeArtQuickFrameInfoGetter (impl) {
 }
 
 function _getArtQuickFrameInfoGetterThunk (impl) {
-  const thunk = makeThunk(64, writer => {
-    writer.putPushRegReg('x0', 'lr');
-    writer.putCallAddressWithArguments(impl, ['x1']);
-    writer.putPopRegReg('x2', 'lr');
-    writer.putStrRegRegOffset('x0', 'x2', 0);
-    writer.putStrRegRegOffset('w1', 'x2', 8);
-    writer.putRet();
-  });
+  if (Process.arch == 'arm64') {
+    const thunk = makeThunk(64, writer => {
+      writer.putPushRegReg('x0', 'lr');
+      writer.putCallAddressWithArguments(impl, ['x1']);
+      writer.putPopRegReg('x2', 'lr');
+      writer.putStrRegRegOffset('x0', 'x2', 0);
+      writer.putStrRegRegOffset('w1', 'x2', 8);
+      writer.putRet();
+    });
+    return new NativeFunction(thunk, 'void', ['pointer', 'pointer'], nativeFunctionOptions);
+  } else if (Process.arch == 'arm') {
+    const thunk = makeThunk(64, writer => {
+      // By call convention, we pass a pointer for the result struct
+      writer.putCallAddressWithArguments(impl, ['r0', 'r1']);
+      writer.putPopRegs(['r0', 'lr']);
+      writer.putMovRegReg('pc', 'lr');
+    });
+    return new NativeFunction(thunk, 'void', ['pointer', 'pointer'], nativeFunctionOptions);
+  } else if (Process.arch == 'ia32') {
+    const thunk = makeThunk(64, writer => {
+      writer.putMovRegRegOffsetPtr('ecx', 'esp', 4); // result
+      writer.putMovRegRegOffsetPtr('edx', 'esp', 8); // self
+      writer.putCallAddressWithArguments(impl, ['ecx', 'edx']);
+      
+      // restore callee's stack frame
+      writer.putMovRegReg('esp', 'ebp');
+      writer.putPopReg('ebp');
+
+      writer.putRet();
+    });
+    return new NativeFunction(thunk, 'void', ['pointer', 'pointer'], nativeFunctionOptions);
+  } else if (Process.arch == 'x64') {
+    const thunk = makeThunk(64, writer => {
+
+      writer.putPushReg('rdi') // preserve result buffer pointer
```

Comment:
Something is off with the indentation here. Did you run this PR through `./node_modules/.bin/semistandard`?

Reply:
I didn't know `semistandard` beforehand. In that line it only reported a missing semicolon, though.

Anyway, `semistandard` reports no problems now.